rocket/src/TLB.scala

```diff
@@ -304,16 +312,40 @@ case class TLBConfig(
   * @param cfg [[TLBConfig]]
   * @param edge collect SoC metadata.
   */
-class TLB(instruction: Boolean, lgMaxSize: Int, cfg: TLBConfig)(implicit edge: TLEdgeOut, p: Parameters) extends CoreModule()(p) {
+class TLB(
+  instruction: Boolean,
+  lgMaxSize: Int,
+  cfg: TLBConfig,
+  edge: TLEdgeOut,
+  pmpGranularity: Int,
+  pgLevels: Int,
+  minPgLevels: Int,
+  pgLevelBits: Int,
+  pgIdxBits: Int,
+  vpnBits: Int,
+  ppnBits: Int,
+  vaddrBits: Int,
+  vaddrBitsExtended: Int,
+  paddrBits: Int,
+  hypervisorExtraAddrBits: Int,
+  asIdBits: Int,
+  xLen: Int,
+  cacheBlockBytes: Int,
+  usingHypervisor: Boolean,
+  usingVM: Boolean,
+  usingAtomics: Boolean,
+  usingAtomicsInCache: Boolean,
+  usingAtomicsOnlyForIO: Boolean,
+  usingDataScratchpad: Boolean) extends Module with MemoryOpConstants {
```

Comment:
it's OK for now, eventually we can use `case class` to package them and use github.com:chipsalliance/chisel#2857 tfor serialization.